src/main/java/graphql/execution/instrumentation/dataloader/DataLoaderDispatcherInstrumentationState.java

```diff
@@ -1,28 +1,27 @@
 package graphql.execution.instrumentation.dataloader;
 
+import graphql.execution.instrumentation.InstrumentationState;
 import org.dataloader.DataLoaderRegistry;
 import org.slf4j.Logger;
 
-import graphql.execution.instrumentation.InstrumentationState;
-
 /**
  * A base class that keeps track of whether aggressive batching can be used
  */
 public class DataLoaderDispatcherInstrumentationState implements InstrumentationState {
 
     private final FieldLevelTrackingApproach approach;
-
     private final DataLoaderRegistry dataLoaderRegistry;
-
     private final InstrumentationState state;
-
+    private final boolean hasNoDataLoaders;
     private boolean aggressivelyBatching = true;
 
     public DataLoaderDispatcherInstrumentationState(Logger log, DataLoaderRegistry dataLoaderRegistry) {
 
         this.dataLoaderRegistry = dataLoaderRegistry;
         this.approach = new FieldLevelTrackingApproach(log, dataLoaderRegistry);
         this.state = approach.createState();
+        hasNoDataLoaders = dataLoaderRegistry.getKeys().isEmpty();
```

Comment:
I see: so instead a nullalble or Optional DRL we check for content here. DLR is immutable, right?

Reply:
right - we always have a  DLR in play